pkg/controller.v1/xgboost/xgboostjob_controller.go

```diff
@@ -325,7 +325,7 @@ func (r *XGBoostJobReconciler) GetPodsForJob(obj interface{}) ([]*corev1.Pod, er
 		return nil, err
 	}
 
-	return util.ConvertPodList(podlist.Items), nil
+	return util.ConvertPodList(util.FilterJobPodList(podlist.Items, job.GetUID())), nil
```

Comment:
Why do we need to create a separate function,`util.FilterJobPodList`?
Should we run the process in the function, `util.ConvertPodList `?

If we do, we can reduce the number of calling loops.

Reply:
good point, I'll fix it.